staging/src/k8s.io/client-go/plugin/pkg/client/auth/azure/ panics if persister is nil

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x1115a97]

goroutine 1 [running]:
k8s.io/client-go/plugin/pkg/client/auth/azure.(*azureTokenSource).storeTokenInCfg(0xc00023bdc0, 0xc000592000, 0xd, 0xc000592000)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/plugin/pkg/client/auth/azure/azure.go:276 +0x397
k8s.io/client-go/plugin/pkg/client/auth/azure.(*azureTokenSource).Token(0xc00023bdc0, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/plugin/pkg/client/auth/azure/azure.go:206 +0x2b3
k8s.io/client-go/plugin/pkg/client/auth/azure.(*azureRoundTripper).RoundTrip(0xc0003533c0, 0xc0001e0600, 0x13e131d, 0xa, 0xc000375d80)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/plugin/pkg/client/auth/azure/azure.go:125 +0x88
k8s.io/client-go/transport.(*userAgentRoundTripper).RoundTrip(0xc0003533e0, 0xc0001e0500, 0xc0003533e0, 0x0, 0x0)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/transport/round_trippers.go:159 +0x1c7
net/http.send(0xc0001e0500, 0x1584520, 0xc0003533e0, 0x0, 0x0, 0x0, 0xc0000cadd8, 0x2030000, 0x1, 0x0)
        /opt/hostedtoolcache/go/1.14.2/x64/src/net/http/client.go:252 +0x445
net/http.(*Client).send(0xc00033f3e0, 0xc0001e0500, 0x0, 0x0, 0x0, 0xc0000cadd8, 0x0, 0x1, 0xc000375a20)
        /opt/hostedtoolcache/go/1.14.2/x64/src/net/http/client.go:176 +0x101
net/http.(*Client).do(0xc00033f3e0, 0xc0001e0500, 0x0, 0x0, 0x0)
        /opt/hostedtoolcache/go/1.14.2/x64/src/net/http/client.go:699 +0x451
net/http.(*Client).Do(...)
        /opt/hostedtoolcache/go/1.14.2/x64/src/net/http/client.go:567
k8s.io/client-go/rest.(*Request).request(0xc00026e960, 0xc0000c7830, 0x0, 0x0)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/rest/request.go:801 +0x3f0
k8s.io/client-go/rest.(*Request).Do(0xc00026e960, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/rest/request.go:873 +0xdf
k8s.io/client-go/kubernetes/typed/core/v1.(*namespaces).List(0xc00036ab50, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/runner/go/pkg/mod/k8s.io/client-go@v0.17.0/kubernetes/typed/core/v1/namespace.go:86 +0x17e
main.queryNamespaces(0xc0000c7e68, 0xc000281408, 0x7, 0xc000281188, 0x7, 0x0)
        /home/runner/work/kubectx/kubectx/cmd/kubens/list.go:73 +0x2b7
main.ListOp.Run(0x1584900, 0xc0000ca008, 0x1584900, 0xc0000ca010, 0x0, 0x0)
        /home/runner/work/kubectx/kubectx/cmd/kubens/list.go:37 +0x209
main.main()
        /home/runner/work/kubectx/kubectx/cmd/kubens/main.go:20 +0x10f
```
**What you expected to happen**:
no panicking 

**How to reproduce it (as minimally and precisely as possible)**:
use azure or OIDC auth plugin with a `clientcmd.RESTConfigFromKubeConfig` or` DirectClientConfig` without a `ConfigAccess (nil)`

**Anything else we need to know?**:
a similar OIDC panic issue is registered here #90766

--- a/staging/src/k8s.io/client-go/rest/plugin.go
+++ b/staging/src/k8s.io/client-go/rest/plugin.go
@@ -47,6 +47,13 @@ type AuthProviderConfigPersister interface {
 	Persist(map[string]string) error
 }
 
+type noopPersister struct{}
+
+func (n *noopPersister) Persist(_ map[string]string) error {
+	// no operation persister
+	return nil
+}
+
 // All registered auth provider plugins.
 var pluginsLock sync.Mutex
 var plugins = make(map[string]Factory)
@@ -68,6 +75,9 @@ func GetAuthProvider(clusterAddress string, apc *clientcmdapi.AuthProviderConfig
 	p, ok := plugins[apc.Name]
 	if !ok {
 		return nil, fmt.Errorf("no Auth Provider found for name %q", apc.Name)
+	}
+	if persister == nil {
+		persister = &noopPersister{}
 	}
 	return p(clusterAddress, apc.Config, persister)
 }

--- a/staging/src/k8s.io/client-go/rest/plugin_test.go
+++ b/staging/src/k8s.io/client-go/rest/plugin_test.go
@@ -171,6 +171,28 @@ func TestAuthPluginPersist(t *testing.T) {
 
 }
 
+func Test_WhenNilPersister_NoOpPersisterIsAssigned(t *testing.T) {
+
+	if err := RegisterAuthProviderPlugin("anyPlugin", pluginPersistProvider); err != nil {
+		t.Errorf("unexpected error: failed to register 'anyPlugin': %v", err)
+	}
+	cfg := &clientcmdapi.AuthProviderConfig{
+		Name:   "anyPlugin",
+		Config: nil,
+	}
+	plugin, err := GetAuthProvider("127.0.0.1", cfg, nil)
+	if err != nil {
+		t.Errorf("unexpected error: failed to get 'anyPlugin': %v", err)
+	}
+
+	anyPlugin := plugin.(*pluginPersist)
+
+	if _, ok := anyPlugin.persister.(*noopPersister); !ok {
+		t.Errorf("expected to be No Operation persister")
+	}
+
+}
+
 // emptyTransport provides an empty http.Response with an initialized header
 // to allow wrapping RoundTrippers to set header values.
 type emptyTransport struct{}

